developer/system/template-manager.md

```diff
@@ -0,0 +1,259 @@
+
+The package name should be in the form `qubes-template-${template_name}`.
+
+For the package metadata (e.g., summary, description), one needs to ensure that the character `|` does not appear.
+This is because we use the character as a separator internally.
+Failing to do so may result in `qvm-template` to fail with cryptic error messages.
+Note that as we already consider the repository metadata untrusted, this should not result in security issues.
+
+The file structure should be quite similar to previous template RPMs.
+Namely, there should be the following files in the package:
+
+- `var/lib/qubes/vm-templates/${template_name}/root.img.part.{00,01,...}`
+  - Split tarball of template `root.img`
+  - Note that the file is still split due to tools such as `rpm2cpio` not supporting large files.
+    (Notably, the cpio format does not support files over 4GiB.)
+- `var/lib/qubes/vm-templates/${template_name}/template.conf`
+  - Stores custom package metadata (as RPM does not support custom attributes).
+  - Uses `KEY=VALUE` format.
+  - Fields (corresponding to [qvm-prefs]/[qvm-features] tags with the same names)
+    - `virt_mode`
+      - Setting this to `pv` requires user confirmation.
+    - `kernel`
+      - Only allowed to be set to "", i.e., "none", for PVGrub.
+    - `no-monitor-layout`
+    - `pci-e820-host`
+    - `linux-stubdom`
+    - `gui`
+    - `gui-emulated`
+    - `qrexec`
+      - These six flags are only allowed to be set to "1" or "0", denoting "true" and "false" respectively.
+    - `net.fake-ip`
+    - `net.fake-gateway`
+    - `net.fake-netmask`
+- `var/lib/qubes/vm-templates/${template_name}/whitelisted-appmenus.list`
+  - Contains default app menu entries of the template itself.
+- `var/lib/qubes/vm-templates/${template_name}/vm-whitelisted-appmenus.list`
+  - Contains default app menu entries of VMs based on the template.
+- `var/lib/qubes/vm-templates/${template_name}/netvm-whitelisted-appmenus.list`
+  - Contains default app menu entries of NetVMs based on the template.
+  - These three files are the same as the current format.
+  - Note that the contents of these files are stored in `qvm-features` upon installation.
+    See the section below for details.
+
+[QSB-28]: https://github.com/QubesOS/qubes-secpack/blob/master/QSBs/qsb-028-2016.txt
+
+[qvm-prefs]: https://dev.qubes-os.org/projects/core-admin-client/en/stable/manpages/qvm-prefs.html#common-properties
+
+[qvm-features]: https://dev.qubes-os.org/projects/core-admin-client/en/stable/manpages/qvm-features.html#list-of-known-features
+
+Metadata Storage
+----------------
+
+The template manager needs to keep metadata of installed templates such as versions and origin.
+This data can be stored via `qvm-features` to keep things consistent when, e.g., `qvm-remove` is used.
+Besides, backups are also more easily handled this way.
+
+Also, the fields can serve as an indicator of whether a template is installed by `qvm-template`.
+
+### Fields ###
```

Comment:
Minor: It's a bit nicer to use the same heading syntax throughout, but this still works.

Reply:
The reason I did this was that my interpretation of [the guidelines] was to use [markdownlint] `setext_with_atx_closed` (i.e., setext for h1 and h2; closed atx for h3 and above). Perhaps I'm understanding it incorrectly?

[the guidelines]: https://www.qubes-os.org/doc/doc-guidelines/#markdown-conventions
[markdownlint]: https://github.com/DavidAnson/markdownlint/blob/main/doc/Rules.md#md003---heading-style

Reply:
Yes. That bullet point is poorly-written. I will fix it.

You want to be consistent with your heading style throughout the document. If you only ever use h1 and h2, then you can simply use setext. However, since you use h3, you must use atx.

Again, this is a minor point, because it will still render correctly. It's just about making the source more consistent.

Reply:
On further reflection, it would probably be better for us just to recommend atx universally.

Reply:
The headings should now all be in atx.